destalinator.py

```diff
@@ -120,18 +120,21 @@ def stale(self, channel_name, days):
             # self.debug("Not checking if {} is stale -- it's too new".format(channel_name))
             return False
         messages = self.get_messages(channel_name, days)
-        messages = [
+        # return True (stale) if none of the messages match the criteria below
+        return not any(
```

Comment:
If we use the not any, moving the if clause to the item clause makes more sense type wise, e.g. 

```python
return not any(
            x.get("user") not in self.config.ignore_users
            and (
               # the message must have text that doesn't include ignored words
                (x.get("text") and b":dolphin:" not in x.get("text").encode('utf-8', 'ignore'))
                # or the message must have attachments
                or x.get("attachments")
            )
       for x in messages
)
```

but I think what you have works.